conda/recipes/rapids_core_dependencies/meta.yaml

```
@@ -16,18 +16,25 @@ build:
   number: {{ GIT_DESCRIBE_NUMBER }}
   string: cuda{{ cuda_major }}_{{ date_string }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
   ignore_run_exports_from:
-    - {{ compiler('cuda') }}
+    {% if cuda_major == "11" %}
+    - {{ compiler('cuda11') }}
+    {% endif %}
 
 requirements:
   build:
-    - cmake>=3.23.1,!=3.25.0
+    - cmake {{ cmake_version }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }} {{ cuda_version }}
+    {% if cuda_major == "11" %}
+    - {{ compiler('cuda11') }} {{ cuda_version }}
+    {% else %}
+    - {{ compiler('cuda') }}
+    {% endif %}
+    - cuda-version ={{ cuda_version }}
     - make
     - ninja
     - sysroot_{{ target_platform }} {{ sysroot_version }}
   host:
-    - cudatoolkit ={{ cuda_version }}
+    - cuda-version ={{ cuda_version }}
```

I think we want a run pinning here too? That ensures that the two packages for 11 and 12 can be distinguished by the user's desired CUDA version.
```suggestion
    - cuda-version ={{ cuda_version }}
    run:
    - {{ pin_compatible('cuda-version', max_pin='x', min_pin='x') }}
```